src/main/java/dev/socketmods/socketnukes/client/render/bolb/BolbHatLayer.java

```diff
@@ -20,37 +17,25 @@
  */
 public class BolbHatLayer extends LayerRenderer<BolbEntity, BolbModel> {
 
-    private static final String CURLE = "Curle";
-
     public BolbHatLayer(IEntityRenderer<BolbEntity, BolbModel> renderManager) {
         super(renderManager);
     }
 
     @Override
     public void render(MatrixStack stack, IRenderTypeBuffer buffers, int packedLight, BolbEntity bolb, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        if (isCurle(bolb) && !bolb.isInvisible()) {
-            IVertexBuilder builder = buffers.getBuffer(RenderType.getEntitySolid(bolb.getEntityTexture()));
-
-            int packedOverlay = LivingRenderer.getPackedOverlay(bolb, 0.0F);
-            float rotationYaw = MathHelper.lerp(partialTicks, bolb.prevRotationYaw, bolb.rotationYaw) -
-                MathHelper.lerp(partialTicks, bolb.prevRenderYawOffset, bolb.renderYawOffset);
-            float rotationPitch = MathHelper.lerp(partialTicks, bolb.prevRotationPitch, bolb.rotationPitch);
-
-            stack.push();
-
-            stack.rotate(Vector3f.YP.rotationDegrees(rotationYaw));
-            stack.rotate(Vector3f.XP.rotationDegrees(rotationPitch));
-            stack.translate(-0.1, 1D, -0.1D);
-            stack.rotate(Vector3f.XP.rotationDegrees(-rotationPitch));
-            stack.rotate(Vector3f.YP.rotationDegrees(-rotationYaw));
-            stack.scale(0.7F, 0.7F, 0.7F);
```

Comment:
I see that in your changes, the rotate calls were removed. Did they not do anything?

Reply:
The opposite they did something.

That code was based on the Deadmau5 Ear Layer, those rotation then translation calls are for aligning the model to the head's current rotation. Bolbs (Slimes) are only really bodies, having the rotation linked to the head was causing graphical glitches where the hat would rotate from the current body (matching the head looking at the new target location i believe) before after a second or so the body would rotate to match the hat as the bolb jumps.

Removing those calls means the model is rendered based on the body which is what we want.

Reply:
Hmm, sounds fine then, since this is only for bolbs (slimes).